json/src/JSON/Parser.idr

```diff
@@ -181,6 +183,14 @@ invalidKey : StrictTok e JSToken
 invalidKey (x::xs) = if isAlpha x then invalidKey xs else unknownRange Same (x::xs)
 invalidKey []      = unknownRange Same []
 
+toToken : Cast String a => (a -> JSON) -> SnocList Char -> JSToken
+toToken toJSON = (Lit . toJSON . cast . cast {to = String})
+
+numberToken : SnocList Char -> JSToken
+numberToken l = case find (== '.') l of
```

Comment:
I think we should not only check for `'.'` but also `'e'` and '`E`' here.

Reply:
Good catch! I added those checks now.